src/patterns.rs

```diff
@@ -271,9 +271,13 @@ impl Rewrite for Pat {
             PatKind::MacCall(ref mac) => {
                 rewrite_macro(mac, None, context, shape, MacroPosition::Pat)
             }
-            PatKind::Paren(ref pat) => pat
-                .rewrite(context, shape.offset_left(1)?.sub_width(1)?)
-                .map(|inner_pat| format!("({})", inner_pat)),
+            PatKind::Paren(ref pat) => {
+                let inner_pat = pat.rewrite(context, shape)?;
+                let pre_snippet = context.snippet(mk_sp(self.span.lo(), pat.span.lo()));
+                let post_snippet = context.snippet(mk_sp(pat.span.hi(), self.span.hi()));
```

Comment:
When I was suggesting we return `context.snippet(span)` I was thinking more along the lines of the following approach where we don't apply any formatting, not even to the inner pattern.

```rust
let pre_snippet = context.snippet(self.span.lo().between(pat.span.lo()));
let post_snippet = context.snippet(pat.span.lo().between(self.span.hi()));

if contains_comment(pre_snippet) || contains_comment(post_snippet) {
    return Some(context.snippet(self.span).to_owned())
}
```

Your approach may also work though! As I mentioned I'd like you to [add some test cases](https://github.com/rust-lang/rustfmt/blob/master/Contributing.md#create-test-cases) to better prove this out.

Reply:
Hey I've added some test cases, let me know if there's anything specific you'd like me to include.